src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextInputFormatter.cs

```diff
@@ -82,23 +129,30 @@ protected Encoding SelectCharacterEncoding(InputFormatterContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (SupportedEncodings?.Count == 0)
+            if (SupportedEncodings.Count == 0)
             {
-                var message = Resources.FormatTextInputFormatter_SupportedEncodingsMustNotBeEmpty(nameof(SupportedEncodings));
+                var message = Resources.FormatTextInputFormatter_SupportedEncodingsMustNotBeEmpty(
+                    nameof(SupportedEncodings));
                 throw new InvalidOperationException(message);
             }
 
-            var request = context.HttpContext.Request;
-
-            var requestEncoding = request.ContentType == null ? null : MediaType.GetEncoding(request.ContentType);
```

Comment:
How could we fix that method or the `MediaType.Encoding` property without breaking back-compatibility? The decision we made was to avoid using the method / property until 2.0.